packages/myst-transforms/src/frontmatter.ts

```diff
@@ -70,7 +77,7 @@ export function getFrontmatter(
   if (nextNodeIsH1 && !titleNull) {
     const title = toText(nextNode.children);
     // Only remove the title if it is the same
-    if (frontmatter.title && frontmatter.title === title) {
+    if (frontmatter.title && frontmatter.title === title && !opts.keepTitleNode) {
```

Comment:
I think an option like this existed before, and I got rid of it a little hastily... Now, we can once again leave the title node in the tree. Useful for `include` files where we almost certainly do not want to remove headings (frontmatter title for those files is totally lost).